internal/ssm/install_test.go

```diff
@@ -0,0 +1,132 @@
+package ssm_test
+
+import (
+	"context"
+	"net/http"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/ProtonMail/gopenpgp/v3/crypto"
+	"github.com/stretchr/testify/assert"
+	"go.uber.org/zap"
+
+	"github.com/aws/eks-hybrid/internal/ssm"
+	"github.com/aws/eks-hybrid/internal/test"
+	"github.com/aws/eks-hybrid/internal/tracker"
+)
+
+func generateKeyPair(t *testing.T) (string, *crypto.Key) {
+	pgp := crypto.PGP()
+	key, err := pgp.KeyGeneration().
+		AddUserId("test", "test@example.com").
+		New().GenerateKey()
+	assert.NoError(t, err)
+
+	armoredPublicKey, err := key.GetArmoredPublicKey()
+	assert.NoError(t, err)
+
+	return armoredPublicKey, key
+}
+
+func generateSignature(t *testing.T, key *crypto.Key, data []byte) []byte {
+	pgp := crypto.PGP()
+	signer, err := pgp.Sign().SigningKey(key).Detached().New()
+	assert.NoError(t, err)
+	signature, err := signer.Sign(data, crypto.Bytes)
+	assert.NoError(t, err)
+	return signature
+}
+
+func TestInstall(t *testing.T) {
+	publicKey, privateKey := generateKeyPair(t)
+	_, wrongPrivateKey := generateKeyPair(t)
+
+	installerData := []byte("#!/bin/bash\necho 'test installer'")
```

Comment:
do we need this? it's ok, but i don't love making the tests depend on having bash available

I think an empty file should work?